File: android/DataBaseTest/app/src/main/java/com/example/a25492/databasetest/MyDatabaseHelper.java
package com.example.a25492.databasetest;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.widget.Toast;

public class MyDatabaseHelper extends SQLiteOpenHelper {
    public static final String CREATE_BOOK="create table Book("
            +"id integer primary key autoincrement,"
            +"author text,"
            +"price real,"  //不精确的双精度浮点型
            +"pages integer,"
            +"name text)";
    private Context mContext;
    public MyDatabaseHelper(Context context, String name, SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
        mContext=context;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        //调用SQLiteDatabase的execSQL()方法执行建表语句
        db.execSQL(CREATE_BOOK);
        Toast.makeText(mContext,"建表成功",Toast.LENGTH_SHORT).show();


    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
//
    }
}
//把建表语句定义成了一个字符串常量,在onCreate()方法中又调用了SQLiteDatabase的execSQL()方法执行建表语句

File: Kotlin/DataBaseText/app/src/main/java/com/example/administrator/databasetext/MyDatabaseHelper.kt
package com.example.administrator.databasetext

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.widget.Toast

class MyDatabaseHelper : SQLiteOpenHelper {
    //建表（数据库）
    val CREATE_BOOK = "create table Book(" +
            "id integer primary key autoincrement," +
            "author text," +
            "price real," +
            "pages integer," +
            "name text)"

    var mContext: Context? = null
        //初始化
    constructor(context: Context, name: String, factory: SQLiteDatabase.CursorFactory?, version: Int) : super(context, name, factory, version) {
        mContext = context
    }

    override fun onCreate(db: SQLiteDatabase?) {
        //执行建表语句
    db?.execSQL(CREATE_BOOK)
        Toast.makeText(mContext,"创建成功",Toast.LENGTH_LONG).show()
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }


}

